backend/app/ingestion/pdf_processor.py

```
@@ -452,6 +453,93 @@ def process_pdf(
+        if pii_field_count > 0:
+            try:
+                masked_md_parts = []
+                for ev in evidence:
+                    if ev.get("content_type") == "text":
+                        fd = ev.get("full_data") or {}
+                        masked_md_parts.append(fd.get("text", ""))
+                if masked_md_parts:
+                    markdown_path.write_text(
+                        "\n\n".join(masked_md_parts), encoding="utf-8"
+                    )
+            except Exception:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)